sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -173,8 +174,10 @@ object DataType {
       DoubleType, FloatType, ShortType, ByteType, StringType, CalendarIntervalType,
       // TODO(SPARK-35732): Parse DayTimeIntervalType from JSON
       DayTimeIntervalType(),
-      // TODO(SPARK-35770): Parse YearMonthIntervalType from JSON
-      YearMonthIntervalType(), TimestampWithoutTZType)
+      YearMonthIntervalType(YEAR, YEAR),
+      YearMonthIntervalType(MONTH, MONTH),
+      YearMonthIntervalType(YEAR, MONTH),
```

Comment:
@cloud-fan You had some concerns of listing all types because we can extend `YearMonthIntervalType`/`DayTimeIntervalType` by precision in the future. So, we cannot list all possible types as we cannot list all possible types of Decimal. WDYT? Could you look at this, please.

Reply:
It's internal code and I'm OK with this. You mentioned a good point before that we can reuse the sql parser here, and we can refactor this part later.